File: backend-scala/src/main/scala/com/example/modules/orders/publicapi/OrdersPublicApi.scala
// ONLY interface other modules/apps may import from the orders module.
package com.example.modules.orders.publicapi

import com.example.modules.orders.internal.{Order, OrderService}

class OrdersPublicApi:
  private val svc = new OrderService
  def createOrder(orderId: String, totalCents: Long): Order = svc.createOrder(orderId, totalCents)
  def getOrder(orderId: String): Option[Order] = svc.getOrder(orderId)
  def markPaid(orderId: String, paymentId: String): Option[Order] = svc.markPaid(orderId, paymentId)
  def markFailed(orderId: String, reason: String): Option[Order] = svc.markFailed(orderId, reason)
  def markRefunded(orderId: String, amountCents: Long): Option[Order] = svc.markRefunded(orderId, amountCents)

File: backend-kotlin/src/main/kotlin/com/example/modules/orders/publicapi/OrdersPublicApi.kt
// ONLY interface other modules/apps may import from the orders module.
package com.example.modules.orders.publicapi

import com.example.modules.orders.internal.OrderService

class OrdersPublicApi {
    private val svc = OrderService()
    fun createOrder(orderId: String, totalCents: Long) = svc.createOrder(orderId, totalCents)
    fun getOrder(orderId: String) = svc.getOrder(orderId)
    fun markPaid(orderId: String, paymentId: String) = svc.markPaid(orderId, paymentId)
    fun markFailed(orderId: String, reason: String) = svc.markFailed(orderId, reason)
    fun markRefunded(orderId: String, amountCents: Long) = svc.markRefunded(orderId, amountCents)
}
